azure/compute/src/main/java/org/jclouds/azurecompute/config/AzureComputeApiModule.java

```diff
@@ -0,0 +1,100 @@
+* limitations under the License.
+*
+* This file has been modified by Microsoft Open Technologies, Inc.
+* Modifications copyright (c) Microsoft Open Technologies, Inc.
+*/
+package org.jclouds.azurecompute.config;
+
+import static org.jclouds.Constants.PROPERTY_SESSION_INTERVAL;
+
+import java.net.Proxy;
+import java.net.URI;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Named;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.inject.AbstractModule;
+import com.google.inject.TypeLiteral;
+import org.jclouds.azurecompute.location.AzureDefaultLocationSupplier;
+import org.jclouds.date.TimeStamp;
+import org.jclouds.azurecompute.AzureComputeApi;
+//import org.jclouds.azurecompute.location.AzureDefaultLocationSupplier;
+import org.jclouds.domain.Location;
+import org.jclouds.functions.IdentityFunction;
+import org.jclouds.internal.FilterStringsBoundToInjectorByName;
+import org.jclouds.json.config.GsonModule;
+import org.jclouds.location.config.LocationModule;
+import org.jclouds.location.suppliers.ImplicitLocationSupplier;
+import org.jclouds.proxy.ProxyForURI;
+import org.jclouds.rest.ApiContext;
+import org.jclouds.rest.ConfiguresRestClient;
+import org.jclouds.rest.config.HttpApiModule;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import org.jclouds.rest.internal.ApiContextImpl;
+
+
+@ConfiguresRestClient
+public class AzureComputeApiModule extends AbstractModule {
+
+   @Provides
+   @TimeStamp
+   protected Long provideTimeStamp(@TimeStamp Supplier<Long> cache) {
+      return cache.get();
+   }
+
+   /**
+    * borrowing concurrency code to ensure that caching takes place properly
+    */
+   @Provides
+   @TimeStamp
+   Supplier<Long> provideTimeStampCache(@Named(PROPERTY_SESSION_INTERVAL) long seconds) {
+      return Suppliers.memoizeWithExpiration(new Supplier<Long>() {
+         public Long get() {
+            return System.currentTimeMillis() / 1000;
+         }
+      }, seconds, TimeUnit.SECONDS);
+   }
+
+   @Override
+   protected void configure() {
+       install(new GsonModule());
+       bind(IdentityFunction.class).toInstance(IdentityFunction.INSTANCE);
+
+       bind(new TypeLiteral<Function<URI, Proxy>>() {
+       }).to(ProxyForURI.class);
+       bind(new TypeLiteral<Function<Predicate<String>, Map<String, String>>>() {
+       }).to(FilterStringsBoundToInjectorByName.class);
+//      install(new GoGridParserModule());
+//      super.configure();
+       bind(new TypeLiteral<ApiContext<AzureComputeApi>>() {
+       }).to(new TypeLiteral<ApiContextImpl<AzureComputeApi>>() {
+       });
+       installLocations();
+   }
+
+   protected void installLocations() {
+      install(new LocationModule());
+      bind(ImplicitLocationSupplier.class).to(AzureDefaultLocationSupplier.class).in(Scopes.SINGLETON);
```

Comment:
All this can be removed. jclouds will provide a default location (the first region) by default.